stan/math/prim/prob/beta_neg_binomial_cdf.hpp

```diff
@@ -0,0 +1,168 @@
+  check_positive_finite(function, "Prior success parameter", alpha_ref);
+  check_positive_finite(function, "Prior failure parameter", beta_ref);
+
+  scalar_seq_view<T_n> n_vec(n);
+  scalar_seq_view<T_r_ref> r_vec(r_ref);
+  scalar_seq_view<T_alpha_ref> alpha_vec(alpha_ref);
+  scalar_seq_view<T_beta_ref> beta_vec(beta_ref);
+  int size_n = stan::math::size(n);
+  size_t max_size_seq_view = max_size(n, r, alpha, beta);
+
+  // Explicit return for extreme values
+  // The gradients are technically ill-defined, but treated as zero
+  for (int i = 0; i < size_n; i++) {
+    if (n_vec.val(i) < 0) {
+      return 0.0;
+    }
+  }
+
+  using T_partials_return = partials_return_t<T_n, T_r, T_alpha, T_beta>;
+  T_partials_return cdf(1.0);
+  auto ops_partials = make_partials_propagator(r_ref, alpha_ref, beta_ref);
+  for (size_t i = 0; i < max_size_seq_view; i++) {
+    // Explicit return for extreme values
+    // The gradients are technically ill-defined, but treated as zero
+    if (n_vec.val(i) == std::numeric_limits<int>::max()) {
+      return 1.0;
+    }
+    auto n_dbl = n_vec.val(i);
+    auto r_dbl = r_vec.val(i);
+    auto alpha_dbl = alpha_vec.val(i);
+    auto beta_dbl = beta_vec.val(i);
+    auto b_plus_n = beta_dbl + n_dbl;
+    auto r_plus_n = r_dbl + n_dbl;
+    auto a_plus_r = alpha_dbl + r_dbl;
+    using a_t = return_type_t<decltype(b_plus_n), decltype(r_plus_n)>;
+    using b_t = return_type_t<decltype(n_dbl), decltype(a_plus_r),
+                              decltype(b_plus_n)>;
+    auto F = hypergeometric_3F2(
+        std::initializer_list<a_t>{1.0, b_plus_n + 1.0, r_plus_n + 1.0},
+        std::initializer_list<b_t>{n_dbl + 2.0, a_plus_r + b_plus_n + 1.0},
+        1.0);
+    auto C = lgamma(r_plus_n + 1.0) + lbeta(a_plus_r, b_plus_n + 1.0)
+             - lgamma(r_dbl) - lbeta(alpha_dbl, beta_dbl) - lgamma(n_dbl + 2.0);
+    auto ccdf = stan::math::exp(C) * F;
+    cdf *= 1.0 - ccdf;
+
+    if constexpr (!is_constant_all<T_r, T_alpha, T_beta>::value) {
+      auto chain_rule_term = -ccdf / (1.0 - ccdf);
+      auto digamma_n_r_alpha_beta = digamma(a_plus_r + b_plus_n + 1.0);
+      T_partials_return dF[6];
+      grad_F32<false, !is_constant<T_beta>::value, !is_constant_all<T_r>::value,
+               false, true, false>(dF, 1.0, b_plus_n + 1.0, r_plus_n + 1.0,
+                                   n_dbl + 2.0, a_plus_r + b_plus_n + 1.0, 1.0,
+                                   precision, max_steps);
+
+      if constexpr (!is_constant<T_r>::value || !is_constant<T_alpha>::value) {
+        auto digamma_r_alpha = digamma(a_plus_r);
+        if constexpr (!is_constant<T_r>::value) {
+          auto partial_lccdf = digamma(r_plus_n + 1.0)
+                               + (digamma_r_alpha - digamma_n_r_alpha_beta)
+                               + (dF[2] + dF[4]) / F - digamma(r_dbl);
+          partials<0>(ops_partials)[i] += partial_lccdf * chain_rule_term;
```

Comment:
imo I think it's find to just have one line for this

Reply:
Same for the other partial accumulations

Reply:
Will move them in one line